types/src/transaction/mod.rs

```diff
@@ -592,16 +592,54 @@ impl From<VMStatus> for TransactionStatus {
     }
 }
 
+#[derive(Clone, Debug, Eq, PartialEq, Ord, PartialOrd, Hash)]
+pub enum GovernanceRole {
+    LibraRoot,
+    TreasuryCompliance,
+    Validator,
+    ValidatorOperator,
+    DesignatedDealer,
+    NonGovernanceRole,
+}
+
+impl GovernanceRole {
+    pub fn from_role_id(role_id: u64) -> Self {
```

Comment:
will the numeric role_id mappings be documented/maintained somewhere?

Reply:
Good question! yes. These are documented in the Move modules published on-chain (specifically Roles.move).